trustme/__init__.py

```diff
@@ -206,6 +203,24 @@ def tempfile(self, dir: Optional[str] = None) -> Generator[str, None, None]:
             os.unlink(f.name)
 
 
+class KeyType(Enum):
+    """Type of the key used to generate a certificate"""
+
+    RSA = 0
+    ECDSA = 1
+
+    def _generate_key(self) -> CERTIFICATE_PRIVATE_KEY_TYPES:
+        if self is KeyType.RSA:
+            # key_size needs to be a least 2048 to be accepted
+            # on Debian and pressumably other OSes
+
+            return rsa.generate_private_key(
+                public_exponent=65537, key_size=2048
+            )
+        elif self is KeyType.ECDSA:  # pragma: no cover
+            return ec.generate_private_key(ec.SECP256R1())
```

Comment:
I'm not sure I'm following, what is not covered is the case where the key is not `KeyType.RSA` nor `KeyType.ECDSA`? Maybe we can we have a `else` clause that raises a ValueError in that case?

Reply:
Do you mean that we should also add a test for that branch or put the no coverage on the else or also add a test for that? I'm a bit hesitant to add such a check since tgis is already encoded in the type. In the current case, mypy will also warn if one of the branches is forgotten, which would no longer be the case.

I don't have strong feelings for either way, let me know what you prefer and I will implement it.

Reply:
Either works, what I want to avoid is having a nocover pragma on a covered branch, which is what we have here, right?

My understanding is that to avoid that we need an else branch, even if mypy knows it will never be reached. Once we have that branch, let's use the simplest way to make all tools happy (nocover/type ignore/an unit test).

How does that sound?